src/components/routing/routeProtectors/SettingsGuard.js

```diff
@@ -1,13 +1,34 @@
-import {Redirect} from "react-router-dom";
+import { useState, useEffect } from "react";
+import { Redirect } from "react-router-dom";
 import PropTypes from "prop-types";
+import { api } from "../../../helpers/api";
 
-export const SettingsGuard = props => {
-    if (localStorage.getItem("token")) {
-        return props.children;
+export const SettingsGuard = (props) => {
+    const [isAuthorized, setIsAuthorized] = useState(null);
+    const accessCode = localStorage.getItem("lobbyAccessCode");
+    useEffect(() => {
+        const checkAuthorization = async () => {
+            const userId = localStorage.getItem("token");
+            const userResponse = await api.get(`/users/${userId}`);
+            const isLeader = userResponse.data.leader;
+            const url = window.location.href;
+            const urlSplit = url.split("/");
+            console.log(isLeader);
+            if (urlSplit[3] === "lobbies" && urlSplit[4] === accessCode && urlSplit[5] === "settings" && isLeader) {
```

Comment:
I think `urlSplit[3] === "lobbies"` and `urlSplit[5] === "settings"` are always true.